components/nexusoperations/completion.go

```diff
@@ -127,11 +129,59 @@ func handleUnsuccessfulOperationError(
 	}
 }
 
+// Adds a NEXUS_OPERATION_STARTED history event and sets the operation state machine to NEXUS_OPERATION_STATE_STARTED.
+// Necessary if the completion is received before the start response.
+func fabricateStartedEventIfMissing(
+	node *hsm.Node,
+	requestID string,
+	operationID string,
+	startTime *timestamppb.Timestamp,
+	links []*commonpb.Link,
+) error {
+	data, err := hsm.MachineData[Operation](node)
+	if err != nil {
+		return err
+	}
+
+	if TransitionStarted.Possible(data) {
+		if operationID == "" {
+			return serviceerror.NewNotFound("operation not found")
+		}
+
+		eventID, err := hsm.EventIDFromToken(data.ScheduledEventToken)
+		if err != nil {
+			return err
+		}
+
+		node.AddHistoryEvent(enumspb.EVENT_TYPE_NEXUS_OPERATION_STARTED, func(e *historypb.HistoryEvent) {
+			e.Attributes = &historypb.HistoryEvent_NexusOperationStartedEventAttributes{
+				NexusOperationStartedEventAttributes: &historypb.NexusOperationStartedEventAttributes{
+					ScheduledEventId: eventID,
+					OperationId:      operationID,
+					RequestId:        requestID,
+				},
+			}
+			e.Links = links
+			if startTime != nil {
+				e.EventTime = startTime
```

Comment:
I'm not sure this is something that's supported or what the implications are. @yycptt, @alexshtin?

Reply:
I will leave this out of this PR for now and add `StartTime` as an attribute in a followup PR.